Sources/PackageGraph/Pubgrub.swift

```diff
@@ -187,6 +201,10 @@ extension PackageRequirement {
         switch (self, other) {
         case (.versionSet(let lhs), .versionSet(let rhs)):
             return lhs.intersection(rhs) == rhs
+        case (.unversioned, .unversioned):
+            return true
```

Comment:
This might not be correct since `unversioned` is always supposed to "win". Can you keep `unversioned` out of this PR? We need to add it carefully to avoid slipping in bugs that can be hard to find.